pydicer/dataset/nnunet.py

```diff
@@ -412,8 +418,234 @@ def prep_label_map_from_one_hot(
 
         return sitk.Resample(label_map, image)
 
-    def prepare_dataset(self):
-        """Coming soon"""
+    def prepare_dataset(self) -> Path:
+        """Prepare the dataset ready for nnUNet training on the file system.
 
-    def generate_training_scripts(self):
-        """Coming soon"""
+        Raises:
+            SystemError: Raised if split_dataset hasn't yet been run.
+            SystemError: Raised if check_structure_names has detected missing structures for
+              patients.
+
+        Returns:
+            Path: The folder in which the nnUNet dataset has been prepared.
+        """
+
+        if len(self.training_cases) == 0:
+            raise SystemError("training_cases are empty, run split_dataset function first.")
+
+        # First check that all cases (in training set) have the structures which are to be learnt
+        df_structures = self.check_structure_names()
+        df_missing = df_structures.data[df_structures.data.isin([0]).any(axis=1)]
+        if len(df_missing) > 0:
+            raise SystemError(
+                "One or more patients are missing structures. Use the "
+                "check_structure_names function to inspect which strucutures are missing and "
+                "correct before proceeding."
+            )
+
+        nnunet_dir = self.nnunet_raw_path.joinpath(f"Task{self.nnunet_id}_{self.nnunet_name}")
+
+        image_tr_path = nnunet_dir.joinpath("imagesTr")
+        image_tr_path.mkdir(exist_ok=True, parents=True)
+
+        image_ts_path = nnunet_dir.joinpath("imagesTs")
+        image_ts_path.mkdir(exist_ok=True, parents=True)
+
+        label_tr_path = nnunet_dir.joinpath("labelsTr")
+        label_tr_path.mkdir(exist_ok=True, parents=True)
+
+        label_ts_path = nnunet_dir.joinpath("labelsTs")
+        label_ts_path.mkdir(exist_ok=True, parents=True)
+
+        df = read_converted_data(self.working_directory, dataset_name=self.dataset_name)
+
+        for pat_id in self.training_cases:
+            df_pat = df[df.patient_id == pat_id]
+            img_row = df_pat[df_pat.modality == self.image_modality].iloc[0]
+            img_dir = Path(img_row.path)
+            img_file = img_dir.joinpath(f"{self.image_modality}.nii.gz")
+            img = sitk.ReadImage(str(img_file))
+
+            target_img_path = image_tr_path.joinpath(f"{pat_id}_0000.nii.gz")
+
+            target_img_path.unlink(missing_ok=True)
+            target_img_path.symlink_to(img_file)
+
+            structure_set_row = df_pat[df_pat.modality == "RTSTRUCT"].iloc[0]
+            structure_set = StructureSet(structure_set_row, self.mapping_id)
+            pat_label_map = self.prep_label_map_from_one_hot(img, structure_set)
+            target_label_path = label_tr_path.joinpath(f"{pat_id}.nii.gz")
+            sitk.WriteImage(pat_label_map, str(target_label_path))
+
+        # for pat_id in test_ids:
```

Comment:
Safe to remove?

Reply:
Thanks @dalmouiee, will remove now and merge :)